internal/db/lens.go

```diff
@@ -13,22 +13,68 @@ package db
 import (
 	"context"
 
+	"github.com/ipfs/go-cid"
+
 	"github.com/sourcenetwork/corekv"
 	"github.com/sourcenetwork/immutable"
+	"github.com/sourcenetwork/immutable/enumerable"
+	"github.com/sourcenetwork/lens/host-go/config/model"
+	"github.com/sourcenetwork/lens/host-go/store"
 
 	"github.com/sourcenetwork/defradb/client"
 	"github.com/sourcenetwork/defradb/errors"
+	"github.com/sourcenetwork/defradb/internal/datastore"
 	"github.com/sourcenetwork/defradb/internal/db/description"
 )
 
-func (db *DB) setMigration(ctx context.Context, cfg client.LensConfig) error {
+// WARNING - This function does not create a txn ctx if one does not exist
+// and so should not be exposed to users.
+func (db *DB) AddLens(ctx context.Context, cfg model.Lens) (cid.Cid, error) {
+	return db.GetLensStore(ctx).Add(ctx, cfg)
+}
+
+// WARNING - This function does not create a txn ctx if one does not exist
+// and so should not be exposed to users.
+func (db *DB) Transform(
+	ctx context.Context,
+	source enumerable.Enumerable[map[string]any],
+	id string,
+) (enumerable.Enumerable[map[string]any], error) {
+	return db.GetLensStore(ctx).Transform(ctx, source, id)
+}
+
+// WARNING - This function does not create a txn ctx if one does not exist
+// and so should not be exposed to users.
+func (db *DB) Inverse(
+	ctx context.Context,
+	source enumerable.Enumerable[map[string]any],
+	id string,
+) (enumerable.Enumerable[map[string]any], error) {
+	return db.GetLensStore(ctx).Inverse(ctx, source, id)
+}
+
+// WARNING - This function does not create a txn ctx if one does not exist
+// and so should not be exposed to users.
+func (db *DB) GetLensStore(ctx context.Context) store.Store {
```

Comment:
todo: Please make this private as it's only used in this package.

Reply:
Handling against comment on `AddLens`